src/blueapi/service/interface.py

```diff
@@ -176,9 +176,14 @@ def get_plan(name: str) -> PlanModel:
     return PlanModel.from_plan(context().plans[name])
 
 
-def get_devices() -> list[DeviceModel]:
+def get_devices(depth: int) -> list[DeviceModel]:
     """Get all available devices in the BlueskyContext"""
-    return [DeviceModel.from_device(device) for device in context().devices.values()]
+    return [
+        model
+        for device in context().devices.values()
+        for model in DeviceModel.from_device_tree(device, depth)
+        if model.protocols
```

Comment:
Prevent any child devices that are not deserializable as have no protocols and so are not registered?